File: src/games/stendhal/common/Debug.java
/* $Id$ */
/***************************************************************************
 *                   (C) Copyright 2003-2010 - Stendhal                    *
 ***************************************************************************
 ***************************************************************************
 *                                                                         *
 *   This program is free software; you can redistribute it and/or modify  *
 *   it under the terms of the GNU General Public License as published by  *
 *   the Free Software Foundation; either version 2 of the License, or     *
 *   (at your option) any later version.                                   *
 *                                                                         *
 ***************************************************************************/
package games.stendhal.common;

/**
 * Gathers all Debug constants in one place.
 *
 * @author mtotz
 */
public class Debug {
	/** if this is enable tileset are loaded on demand. */
	public static final boolean VERY_FAST_CLIENT_START = true;

	/** enables cycling through the panel textures by clicking at the title bar. */
	public static final boolean CYCLE_PANEL_TEXTURES = false;

	/** version. */
	// Note: This line is updated by build.xml using a regexp so be sure to adjust it in case you modify this line.
	public static final String VERSION = "1.41";

	/** pre release suffix */
	// Note: This line is updated by build.xml using a regexp so be sure to adjust it in case you modify this line.
	public static final String PRE_RELEASE_VERSION = null;

	/**
	 * Log list sizes to find the memory leak. It must be somewhere...
	 */
	public static final boolean SHOW_LIST_SIZES = false;

	/**
	 * it shows entity's view area
	 */
	public static final boolean SHOW_ENTITY_VIEW_AREA = false;
}

File: src/js/stendhal/util/Debug.ts
/***************************************************************************
 *                    Copyright © 2024 - Faiumoni e. V.                    *
 ***************************************************************************
 ***************************************************************************
 *                                                                         *
 *   This program is free software; you can redistribute it and/or modify  *
 *   it under the terms of the GNU Affero General Public License as        *
 *   published by the Free Software Foundation; either version 3 of the    *
 *   License, or (at your option) any later version.                       *
 *                                                                         *
 ***************************************************************************/


export class Debug {

	/** Properties marked for debugging. */
	private static readonly properties: Record<string, boolean> = {};


	/**
	 * Static class.
	 */
	private constructor() {
		// do nothing
	}

	/**
	 * Checks if a property is marked for debugging.
	 *
	 * @param {string} prop
	 *   Property name.
	 * @returns {boolean}
	 *   `true` if `prop` is contained in active list.
	 */
	static isActive(prop: string): boolean {
		return Debug.properties[prop] === true;
	}

	/**
	 * Marks or unmarks a property for debugging.
	 *
	 * @param {string} prop
	 *   Property name.
	 * @param {boolean} active
	 *   `true` to mark as active, `false` for inactive.
	 */
	static setActive(prop: string, active: boolean) {
		Debug.properties[prop] = active;
	}

	/**
	 * Toggles marking property for debugging.
	 *
	 * @param {string} prop
	 *   Property name.
	 * @returns {boolean}
	 *   Property debugging state (`true` if active).
	 */
	static toggle(prop: string): boolean {
		Debug.setActive(prop, !Debug.isActive(prop));
		return Debug.isActive(prop);
	}
}
